src/uct/ib/rc/base/rc_iface.c

```diff
@@ -558,13 +558,23 @@ UCS_CLASS_INIT_FUNC(uct_rc_iface_t, uct_iface_ops_t *tl_ops,
                                                      "RETRY_COUNT",
                                                      config->tx.retry_count,
                                                      UCT_RC_QP_MAX_RETRY_COUNT);
-    self->config.max_rd_atomic  = config->max_rd_atomic;
     self->config.ooo_rw         = config->ooo_rw;
 #if UCS_ENABLE_ASSERT
     self->tx.in_pending         = 0;
 #endif
     max_ib_msg_size             = uct_ib_iface_port_attr(&self->super)->max_msg_sz;
 
+    if (config->max_rd_atomic == UCS_ULUNITS_AUTO) {
```

Comment:
I think that we should handle `INF` value too here, which means the maximal possible value.
```suggestion
    if ((config->max_rd_atomic == UCS_ULUNITS_AUTO) ||
        (config->max_rd_atomic == UCS_ULUNITS_INF)) {
```

Reply:
i'd fail, because maximum allowed is meant by auto, infinite has another meaning.
it is currently fails, but can add more clear error message